fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -185,6 +185,12 @@ def reset_all
         all.each(&:reset)
       end
 
+      def reset_all_by_version(os_version: nil)
+        return false unless os_version
+        version_sim = all.select { |device| device.os_version == os_version }
+        version_sim.each(&:reset) if version_sim
```

Comment:
I think the `if version_sim` is unnecessary here, as `select` should always at least return an empty array. If it is, Let's just do this on one line without creating a temporary var:

`all.select { |device| device.os_version == os_version }.each(&:reset)`